apps/docs/src/guide/introduction/creating-a-fuel-dapp.md

```diff
@@ -0,0 +1,179 @@
+
+```md
+my-fuel-project
+├── src
+│ ├── pages
+│ │ ├── index.tsx
+│ │ └── ...
+│ ├── components
+│ │ └── ...
+│ ├── styles
+│ │ └── ...
+│ └── lib.ts
+├── public
+│ └── ...
+├── sway-programs
+│ ├── contract
+│ │ └── src
+│ │ └── main.sw
+│ ├── Forc.lock
+│ └── Forc.toml
+├── fuels.config.ts
+├── package.json
+└── ...
+```
+
+It is a Next.js project with a few extra files and folders. Let's take a closer look at some of the important ones:
+
+### `./fuels.config.ts`
+
+This is the configuration file for the [`fuels` CLI](../fuels/index.md), the CLI and tooling that powers this project under the hood. It makes sure that all of your Sway programs are continuously compiled and deployed to your local Fuel node. You can read more about the `fuels.config.ts` file in the [Fuels CLI documentation](../fuels/config-file.md).
+
+### `./sway-programs/contract/src/main.sw`
+
+This is where our Sway contract lives. Out of the box, it is a simple counter contract that can only be incremented. We will add a decrement functionality to it in the next step.
+
+### `./src/pages/index.tsx`
+
+This file contains the source code for the frontend of our dApp. It is a Next.js page that renders the counter value and allows the user to increment the counter.
+
+### Dev Environment Setup
+
+Now that we have our project scaffolded, let's set up our development environment.
+
+Let's first start our Fuel Dev server. This will start a local Fuel node and continuously compile and deploy our Sway programs to it.
+
+::: code-group
+
+```sh [npm]
+pnpm fuels:dev
+```
+
+```sh [pnpm]
+pnpm fuels:dev
+```
+
+:::
+
+Once the server is up and running, we can start our Next.js development server in another terminal.
+
+::: code-group
+
+```sh [npm]
+pnpm dev
+```
+
+```sh [pnpm]
+pnpm dev
+```
+
+:::
+
+You should now be able to see the counter dApp running at `http://localhost:3000`. You can try changing the contents of the `./sway-programs/contract/src/main.sw` file and see the changes reflected in the UI without having to restart the server.
+
+![Fullstack Fuel Dev Workflow](../../public/creating-a-fuel-dapp-create-fuels-split-view.png)
+
+## Adding Decrement Functionality
+
+To add decrement functionality to our counter, we will modify the `./sway-programs/contract/src/main.sw` file.
+
+There are two steps when adding a new function to a Sway program:
+
+### 1. Specifying the function's ABI
+
+Towards the top of the file, you will find the ABI section for the contract. Let's add a new function to it:
+
+<<< @/../../create-fuels-counter-guide/sway-programs/contract/src/main.sw#create-fuels-counter-guide-abi{rust:line-numbers}
+
+### 2. Implementing the function
+
+We will add the implementation of the `decrement_counter` function right below the `increment_counter` function.
+
+<<< @/../../create-fuels-counter-guide/sway-programs/contract/src/main.sw#create-fuels-counter-guide-impl{rust:line-numbers}
```

Comment:
IMO this should be split into FE and BE - implement the program then the dApp.

Reply:
Great feedback. I added it here: https://github.com/FuelLabs/fuels-ts/pull/1965/commits/448ca705f41354dda10d436cc2ba37579c3e3b29